src/client/sw/worker.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { on } from '@americanexpress/one-service-worker';
+
+import {
+  createInstallMiddleware,
+  createActivateMiddleware,
+} from './middleware';
+
+try {
+  on('install', createInstallMiddleware());
+
+  on('activate', createActivateMiddleware());
+} catch (e) {
+  // eslint-disable-next-line no-console
+  console.error(e);
```

Comment:
There's a limitation with using asynchronous actions here in the service worker script body. With events, we can use `event.waitUntil` to wait for a `fetch` call for instance or an async write to `indexedDB`. It would be best to message the main thread and have it report the error there - the issue with that is identifying the open client to transfer the message to is an async process. This will take some thinking/experimenting to understand how to best handle errors in this situation. For example, we can dispatch an `error` event a set up a default handler to `waitUntil` we send out the error back to the server.

Reply:
I've used a message handler here to dispatch to main thread, that way we can log the error.